dev/com.ibm.websphere.appserver.features/visibility/public/restfulWSClient-3.1/io.openliberty.restfulWSClient-3.1.feature

```diff
@@ -25,6 +25,7 @@ IBM-ShortName: restfulWSClient-3.1
 WLP-AlsoKnownAs: jaxrsClient-3.1
 Subsystem-Name: Jakarta RESTful Web Services 3.1 Client
 -features=io.openliberty.cdi-4.0, \
+  io.openliberty.jakarta.concurrency-3.0, \
```

Comment:
Performance team will likely object to concurrency-3.0 being automatically enabled by your feature.  They did when John added similar changes in EE 8.  You could switch over to the solution (CompletionStageFactory) that was written for that purpose,

With the CompletionStageFactory, instead of having a direct dependency on Concurrency/ManagedExecutorService,
```
        if (executor instanceof ManagedExecutorService) {
            return ((ManagedExecutorService)executor).supplyAsync(supplier);
        }
```
you would instead delegate to the CompletionStageFactory, supplying the executor,
```
        completionStageFactory.supplyAsync(supplier, executor);
```
But it looks like some of your code has another step where it wants to delegate AsyncClientExecutorService if not a ManagedExecutorService, so the above won't work in those cases.  You could possibly copy the first couple of lines out of the CompletionStageFactory supplyAsync method and do,
```
        if (executor instanceof CompletionStageExecutor)
            return ((CompletionStageExecutor) executor).supplyAsync(supplier);
```

The other option would be isolating the integration with concurrent-3.0 under an autofeature.

Reply:
Thanks Nathan, I believe I've made the changes you suggested.
@jhanders34 Please ensure you concur.  Thanks

Reply:
It was the way it was on purpose.  The concurrent-3.0 feature is NOT enabled with this change.  It is only the API feature which doesn't enable anything except allow restfulWSClient-3.1 to import the class in its bundle and not get a ClassNotFoundException / NoClassDefFoundError.  The reason that I did this was because in Concurrency 3.0 supplyAsync is part of the API now, so we can reference it directly and we can eventually send this code over to the RestEasy team to consume since we are not using internal Liberty APIs.

Reply:
Actually I changed my mind.  Going to have to do it different before sending to RestEasy anyway.

Reply:
sounds good.